mlrun/utils/helpers.py

```diff
@@ -270,6 +271,15 @@ def validate_artifact_key_name(
     )
 
 
+def validate_inline_artifact_body_size(body: str) -> None:
+    body_length = len(body.encode("utf-8"))
+    if body_length > MYSQL_MEDIUMBLOB_BYTES:
+        raise mlrun.errors.MLRunPreconditionFailedError(
+            "The body of the artifact exceeds the maximum allowed size. "
+            "Please reduce the size of the artifact body and try again."
+        )
```

Comment:
I would go with BadRequest
and I think we should discourage using inline artifacts anyway and instead of asking to reduce, perhaps encourage the clients to not used embed artifacts.

```suggestion
        raise mlrun.errors.MLRunBadRequestError(
            "The body of the artifact exceeds the maximum allowed size. "
            "Avoid embedding the artifact body. This increases the size of the project yaml file and could affect the project during loading and saving. More information is available at https://docs.mlrun.org/en/latest/projects/automate-project-git-source.html#setting-and-registering-the-project-artifacts"
        )
```